codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/DocumentationConverter.java

```diff
@@ -133,18 +169,56 @@ private void writeText(TextNode node) {
 
             // Docs can have valid $ characters that shouldn't run through formatters.
             String text = node.text().replace("$", "$$");
-
             if (shouldStripPrefixWhitespace) {
                 shouldStripPrefixWhitespace = false;
                 text = StringUtils.stripStart(text, " \t");
             }
 
-            if (needsListPrefix) {
-                needsListPrefix = false;
-                writer.write("");
-                writeIndent();
-                text = "* " + StringUtils.stripStart(text, " \t");
+            String curString = writer.toString();
+            if (listDepth > 0) {
+                if (needsListPrefix) {
+                    needsListPrefix = false;
+                    StringBuilder sb = new StringBuilder();
+                    for (int i = 0; i < listDepth; i++) {
+                        sb.append("    ");
+                    }
+                    text = "- " + StringUtils.stripStart(text, " \t");
+                    text = sb.append(text).toString();
+                    writeNewline();
+                } else if (curString.charAt(curString.length() - 1) != ' ') {
+                    text = " " + StringUtils.stripStart(text, " \t");
+                } else {
+                    text = StringUtils.stripStart(text, " \t");
+                }
+            } else {
+                // check the last line's remaining space
+                int lastLineRemaining = docWrapLength - (curString.length() - curString.lastIndexOf("\n"));
+                if (lastLineRemaining <= 0) {
+                    writeNewline();
+                    text = StringUtils.wrap(text, docWrapLength);
+                    writer.writeInline(text);
+                    return;
+                }
+
+                if (lastLineRemaining >= text.length()) {
```

Comment:
We definitely need some sort of explanation on the different cases you're handling here (in comment form).